### PR TITLE
S05 us50 select class

### PR DESCRIPTION
**Pull request by cyrius83600**

### CODE REVIEW

File: src/app/pages/classes-page/classes-page.component.scss [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+    border-radius: 30px;
+    background-color:#DDB375;
+    color: black;
+    display:flex;
+    align-items: center;
+    justify-content:center;
+}
+
+.text-description{
+    margin-left:5%;
```
Comment (OCR-VH-DevWeb) [inline src/app/pages/classes-page/classes-page.component.scss:15]: Minify this declaration with margin: 2% 5% 0 5% => 
clockwise direction =>  margin: top right bottom left;

File: src/app/pages/classes-page/classes-page.component.scss [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+    flex-direction: row;
+    justify-content: space-around;
+    padding-bottom:7%;
+}
+
+.selectClass{
+    background-color: #202229;
+    filter: opacity(90%);
+    margin: $globalPaddingTop 0 2rem 0;
+    border-radius: 10px;
```
Comment (OCR-VH-DevWeb) [inline src/app/pages/classes-page/classes-page.component.scss:55]: use the appropriate variable in _import.scss $textColor: black;
Replace color with variable => color :  $textColor

File: src/app/pages/classes-page/classes-page.component.scss [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+    color: white;
```
Comment (OCR-VH-DevWeb) [inline src/app/pages/classes-page/classes-page.component.scss:56]: use the appropriate variable in _import.scss  $textLightColor: #ffffff; 
Replace color with viarable => color:  $textLightColor

File: src/app/pages/classes-page/classes-page.component.scss [hunk 4]
```diff
@@ -0,0 +1,222 @@
...
+    font-size: 20px;
+}
+
+.selectStats {
+    background-color: #202229;
+            filter: opacity(90%);
+            margin-top: $globalPaddingTop;
+            margin-bottom: 2rem;
+            border-radius: 10px;
+            color: white;
```
Comment (OCR-VH-DevWeb) [inline src/app/pages/classes-page/classes-page.component.scss:92]: Same : 
use the appropriate variable in _import.scss  $textLightColor: #ffffff; 
Replace color with viarable =>  color:  $textLightColor

File: src/app/pages/classes-page/classes-page.component.scss [hunk 5]
```diff
@@ -0,0 +1,222 @@
...
+    height: 30px;
+    width: 30px;
+    border-radius: 30px;
+    background-color:#DDB375;
+    color: black;
+    display:flex;
+    align-items: center;
+    justify-content:center;
+    margin-top:8%;
+    justify-content: center;
```
Comment (OCR-VH-DevWeb) [inline src/app/pages/classes-page/classes-page.component.scss:30]: you repet the "justify-content: center;", remove repetition please

File: src/app/pages/classes-page/classes-page.component.scss [hunk 6]
```diff
@@ -0,0 +1,222 @@
...
+        margin-left: 5vw;
+    }
+}
+
+.modifying
+{
+    width:1000px;
+    height:700px;
+    background-color:#202229;
+    margin-top: 7%;
```
Comment (OCR-VH-DevWeb) [inline src/app/pages/classes-page/classes-page.component.scss:126]: Minify this declaration with margin: 2% 5% 0 5% => 
clockwise direction =>  margin: top right bottom left;

File: src/app/pages/classes-page/classes-page.component.scss [hunk 7]
```diff
@@ -0,0 +1,222 @@
...
+            margin-left: 8%;
+            text-align: center;
+            padding: 1rem;
+            padding-bottom: 2rem;
+            width: 25vw;
+}
+
+@media screen and (max-width: 768px) {
+    .selectStats {
+        margin-top: 100px;
```
Comment (OCR-VH-DevWeb) [inline src/app/pages/classes-page/classes-page.component.scss:102]: Minify this declaration with margin: 100px 0 5% 5%;
clockwise direction =>  margin: top right bottom left;

File: src/app/pages/classes-page/classes-page.component.scss (hunk 4 above)
Comment (OCR-VH-DevWeb) [inline src/app/pages/classes-page/classes-page.component.scss:92]: Use the appropriate variable in _import.scss $textLightColor: #ffffff;
Replace color with viarable => color: $textLightColor

File: src/app/pages/classes-page/classes-page.component.scss [hunk 8]
```diff
@@ -0,0 +1,222 @@
...
+.selectStats {
+    background-color: #202229;
+            filter: opacity(90%);
+            margin-top: $globalPaddingTop;
+            margin-bottom: 2rem;
+            border-radius: 10px;
+            color: white;
+            margin-left: 8%;
+            text-align: center;
+            padding: 1rem;
```
Comment (OCR-VH-DevWeb) [inline src/app/pages/classes-page/classes-page.component.scss:95]: You declare global padding with one value and after padding-bottom!
Refactor this well

File: src/app/pages/classes-page/classes-page.component.scss [hunk 9]
```diff
@@ -0,0 +1,222 @@
...
+    .selectClass {
+        width: 90vw;
+        margin-right: 5vw;
+        margin-left: 5vw;
+    }
+}
+
+.modifying
+{
+    width:1000px;
```
Comment (OCR-VH-DevWeb) [inline src/app/pages/classes-page/classes-page.component.scss:123]: That's big value in pixel are you sure?

File: src/app/pages/classes-page/classes-page.component.scss [hunk 10]
```diff
@@ -0,0 +1,222 @@
...
+    font-size:20px;
+}
+
+.validationButton
+{
+    justify-content: center;
+    align-items:center;
+    text-align: center;
+    background-color: #DDB375;
+    color:black;
```
Comment (OCR-VH-DevWeb) [inline src/app/pages/classes-page/classes-page.component.scss:141]: Same as before with the appropriate variable :
Replace color with variable => color : $textColor

File: src/app/pages/classes-page/classes-page.component.scss [hunk 11]
```diff
@@ -0,0 +1,222 @@
...
+.modifyScore
+{
+    text-align: center;
+    margin-top : 2%;
+}
+
+.stat{
+    display:flex;
+    flex-direction:row;
+    color:white;
```
Comment (OCR-VH-DevWeb) [inline src/app/pages/classes-page/classes-page.component.scss:185]: again Replace color with variable => color : $textLightColor

File: src/app/pages/classes-page/classes-page.component.scss [hunk 12]
```diff
@@ -0,0 +1,222 @@
...
+.modifying
+{
+    width:760px;
+    height:600px;
+    background-color:#202229;
+    margin-bottom:12%;
+    margin-left:6%;
+    margin-right:6%;
+    border-radius: 10px;
+    color:white;
```
Comment (OCR-VH-DevWeb) [inline src/app/pages/classes-page/classes-page.component.scss:197]: again Replace color with variable => color : $textLightColor

File: src/app/pages/classes-page/classes-page.component.scss [hunk 13]
```diff
@@ -0,0 +1,222 @@
...
+    flex-direction:row;
+    color:white;
+    height: 40px;
+}
+.modifying
+{
+    width:760px;
+    height:600px;
+    background-color:#202229;
+    margin-bottom:12%;
```
Comment (OCR-VH-DevWeb) [inline src/app/pages/classes-page/classes-page.component.scss:193]: Same as before:
Minify your margin declaration

File: src/app/pages/world-map/world-map.component.scss [hunk 14]
```diff
@@ -0,0 +1,98 @@
+@use "../../../import" as *;
+
+.world_map_container {
+    background-color: #202229;
+    filter: opacity(90%);
+    margin: $globalPaddingTop 0 2rem 0;
+    border-radius: 10px;
+    color: white;
```
Comment (OCR-VH-DevWeb) [inline src/app/pages/world-map/world-map.component.scss:8]: Use the variable color from -import.scss with variable => color : $textLightColor

**Review by OCR-VH-DevWeb — COMMENTED** [review]
Need to make changes for validation

File: src/app/pages/world-map/world-map.component.scss (hunk 14 above)
Comment (cyrius83600) [reply src/app/pages/world-map/world-map.component.scss:8]: This folder must not appear in that branch. It's a mistake. I deleted it.

File: src/app/pages/classes-page/classes-page.component.scss [hunk 15]
```diff
@@ -0,0 +1,222 @@
...
+    .selectStats {
+        margin-top: 100px;
+        width: 90vw;
+        margin-right: 5vw;
+        margin-left: 5vw;
+    }
+    .content-page{
+        justify-content: center;
+        align-items:center;
+        display:flex;
```
[the remark this answers is not in the document]
Comment (cyrius83600) [reply src/app/pages/classes-page/classes-page.component.scss:107]: What must i change ?

[the remark this answers is not in the document]
Comment (OCR-VH-DevWeb) [reply src/app/pages/classes-page/classes-page.component.scss:107]: No change really, just care about declare display flex before the other for the priority of css rules

File: src/app/pages/classes-page/classes-page.component.scss [hunk 16]
```diff
@@ -0,0 +1,217 @@
...
+.selectClass{
+    background-color: #202229;
+    filter: opacity(90%);
+    margin: $globalPaddingTop 0 2rem 0;
+    border-radius: 10px;
+    color: $textLightColor;
+    margin-right: 20%;
+    margin-left: 8%;
+    text-align: center;
+    padding: 1rem;
```
Comment (OCR-VH-DevWeb) [inline src/app/pages/classes-page/classes-page.component.scss:58]: Same as before you must choose between padding: x x x x ; and padding-right etc

File: src/app/pages/classes-page/classes-page.component.scss [hunk 17]
```diff
@@ -0,0 +1,217 @@
...
+
+.content
+{
+    font-size: 20px;
+}
+
+.selectStats {
+    background-color: #202229;
+            filter: opacity(90%);
+            margin-top: $globalPaddingTop;
```
Comment (OCR-VH-DevWeb) [inline src/app/pages/classes-page/classes-page.component.scss:87]: same comment than before bad practice!

**Review by OCR-VH-DevWeb — COMMENTED** [review]
I didn't see all corections than i preconised yesterday evening. Take look of yours declarations of margin and use variables of _import.scss. Thanks

**Review by OCR-VH-DevWeb — CHANGES_REQUESTED** [review]
you didn't process all the detailed comments

File: src/app/pages/classes-page/classes-page.component.ts [hunk 18]
```diff
@@ -0,0 +1,126 @@
...
+    this.points = 5;
+    this.selectedClass = new Specialization();
+    this.apiService.getSpecialization().subscribe(
+      classes => 
+        {
+          this.classes = classes;
+          for(let i = 0; i < classes.length; i++)
+          {
+            const classe = classes[i];
+            var description = classe.content;
```
Comment (OCR-VH-DevWeb) [inline src/app/pages/classes-page/classes-page.component.ts:96]: var is depreciated

File: src/app/pages/classes-page/classes-page.component.html [hunk 19]
```diff
@@ -0,0 +1,54 @@
...
+      <div class="selectClass">
+          <h2 class="title">Sélection des classes</h2>
+            <div class="flexClasses">
+              <button class="content" class="buttonLeft" (click)="decrementCount()"><div class="circleClasses"><</div></button>
+                
+                <ul class="displayPic">
+                @for(filters of filterClasses; track filters)
+                {
+                  <li class="flexPic">
+                    <button  (click)="changePic(filters.name)"><img class="buttonPic" [ngStyle] = "{'border-color': selectedClass.name === filters.name ? '#DDB375' : 'black'}" src="{{filters.url}}"/></button>
```
Comment (OCR-VH-DevWeb) [inline src/app/pages/classes-page/classes-page.component.html:30]: Use the variable $textColor to replace 'black'

File: src/app/pages/classes-page/classes-page.component.ts [hunk 20]
```diff
@@ -0,0 +1,131 @@
...
+  ngOnInit()
+  {
+    this.breakpointObserver.observe(Object.values(this.breakpoints)).subscribe((state: BreakpointState) => {
+    this.isMobile = state.breakpoints[this.breakpoints.mobile]});
+    this.points = 5;
+    this.selectedClass = new Specialization();
+    this.specializationService.getAllSpecialization().subscribe(x=>{
+      this.classes=x});
+  }
+    // this.apiService.getSpecialization().subscribe(
```
Comment (OCR-VH-DevWeb) [inline src/app/pages/classes-page/classes-page.component.ts:94]: delete dead code

**Review by OCR-VH-DevWeb — APPROVED** [review]
Approved

**Review by OCR-VH-DevWeb — APPROVED** [review]
Approved

**Review by Benja-san — APPROVED** [review]

**Pull request merged**